packages/flutter_tools/lib/src/android/android_workflow.dart

```diff
@@ -23,13 +26,74 @@ class AndroidWorkflow extends DoctorValidator implements Workflow {
   @override
   bool get canLaunchDevices => androidSdk != null && androidSdk.validateSdkWellFormed().isEmpty;
 
+  static const String _kJavaHomeEnvironmentVariable = 'JAVA_HOME';
+  static const String _kJavaExecutable = 'java';
+  static const String _kJdkDownload = 'https://www.oracle.com/technetwork/java/javase/downloads/';
+
+  /// First sniff JAVA_HOME, then fallback to PATH.
+  String _findJavaBinary() {
+
+    final String javaHomeEnv = platform.environment[_kJavaHomeEnvironmentVariable];
+    if (javaHomeEnv != null) {
+      // Trust JAVA_HOME.
+      return fs.path.join(javaHomeEnv, 'bin', 'java');
```

Comment:
As long as this string is executed by the process manager we should be good on Windows (the process manager abstracts away the fact that an executable might have a different extensions on Windows. Looks like this string is passed to `processManager.runSync` below, so we should be good here.